tools/functional-tester/etcd-tester/stresser.go

```diff
@@ -71,49 +71,63 @@ func (st *stressTable) choose() stressFunc {
 	return st.entries[idx].f
 }
 
-func newStressPut(kvc pb.KVClient, keySuffixRange, keySize int) stressFunc {
+func (s *stresser) newStressPut(kvc pb.KVClient) stressFunc {
 	return func(ctx context.Context) error {
 		_, err := kvc.Put(ctx, &pb.PutRequest{
-			Key:   []byte(fmt.Sprintf("foo%d", rand.Intn(keySuffixRange))),
-			Value: randBytes(keySize),
+			Key:   []byte(fmt.Sprintf("foo%d", rand.Intn(s.keySuffixRange))),
+			Value: randBytes(s.keySize),
 		}, grpc.FailFast(false))
 		return err
 	}
 }
 
-func newStressRange(kvc pb.KVClient, keySuffixRange int) stressFunc {
+func (s *stresser) newStressRange(kvc pb.KVClient) stressFunc {
 	return func(ctx context.Context) error {
 		_, err := kvc.Range(ctx, &pb.RangeRequest{
-			Key: []byte(fmt.Sprintf("foo%d", rand.Intn(keySuffixRange))),
+			Key: []byte(fmt.Sprintf("foo%d", rand.Intn(s.keySuffixRange))),
 		}, grpc.FailFast(false))
 		return err
 	}
 }
 
-func newStressRangePrefix(kvc pb.KVClient, keySuffixRange int) stressFunc {
+func (s *stresser) newStressRangeInterval(kvc pb.KVClient) stressFunc {
 	return func(ctx context.Context) error {
 		_, err := kvc.Range(ctx, &pb.RangeRequest{
 			Key:      []byte("foo"),
-			RangeEnd: []byte(fmt.Sprintf("foo%d", rand.Intn(keySuffixRange))),
+			RangeEnd: []byte(fmt.Sprintf("foo%d", s.keySuffixRange)),
+			Limit:    int64(s.keyRangeLimit),
 		}, grpc.FailFast(false))
 		return err
 	}
 }
 
-func newStressDelete(kvc pb.KVClient, keySuffixRange int) stressFunc {
+func (s *stresser) newStressDelete(kvc pb.KVClient) stressFunc {
 	return func(ctx context.Context) error {
 		_, err := kvc.DeleteRange(ctx, &pb.DeleteRangeRequest{
-			Key: []byte(fmt.Sprintf("foo%d", rand.Intn(keySuffixRange))),
+			Key: []byte(fmt.Sprintf("foo%d", rand.Intn(s.keySuffixRange))),
 		}, grpc.FailFast(false))
 		return err
 	}
 }
 
-func newStressDeletePrefix(kvc pb.KVClient, keySuffixRange int) stressFunc {
+func (s *stresser) newStressDeleteInterval(kvc pb.KVClient) stressFunc {
 	return func(ctx context.Context) error {
+		resp, _ := kvc.Range(ctx, &pb.RangeRequest{
+			Key:       []byte("foo"),
+			RangeEnd:  []byte(fmt.Sprintf("foo%d", s.keySuffixRange)),
+			SortOrder: pb.RangeRequest_ASCEND,
```

Comment:
the keys will be ascend sorted by key name by default, iirc. Adding a `SortOrder` forces the `Range` to get the full range from the backend before applying the limit